sdcm/tester.py

```diff
@@ -382,14 +404,37 @@ def argus_test_run(self):
             ).publish_or_dump()
             return unittest.mock.MagicMock()
 
+    def start_argus_heartbeat_thread(self) -> threading.Event:
+        def send_argus_heartbeat(connection: ArgusSCTClient, stop_signal: threading.Event):
+            if isinstance(connection, unittest.mock.MagicMock):
+                return
+            fail_count = 0
+            while not stop_signal.is_set():
+                if fail_count > 5:
```

Comment:
Is that what was the logic before ?

Why would we stop trying ? cause argus might be down ?

Anyhow what information do we get out of those heartbeats ? what do we loose if well take them off ?

Reply:
> Is that what was the logic before ?

Without the fail count

> Why would we stop trying ? cause argus might be down ?

In case argus is down / not responding to not spam logs with failed heartbeats.

> Anyhow what information do we get out of those heartbeats ? what do we loose if well take them off ?

Heartbeat timestamp is presented to the user when test is running, absence of it could indicate a crash or another problem.

Reply:
> Without the fail count

To elaborate, it didn't have any exception handling, so the thread would exit on the first heartbeat failure.

Reply:
I'm not sure anyone actually uses that information as part of debugging jobs, most people are unaware of it.

From my POV we can remove it, it doesn't yet add too much value (in less we clearly state to the user that the job is "stuck", which we don't now)

Reply:
> From my POV we can remove it, it doesn't yet add too much value (in less we clearly state to the user that the job is "stuck", which we don't now)

I think we should raise this question with the people before we remove it, and for improvement we could make frontend display a warning when a job heartbeat becomes too outdated - say 5 minutes. 

Additionally, there's another improvement in progress right now to clean up stuck jobs in argus that will rely on this field, so we can't just remove it.

Reply:
o.k. as a user it's a bit meaningless at least to me, that's why I was asking...

but if there other things going to be building ontop of it, that'sf fine...